torch/lib/c10d/Utils.cpp

```diff
@@ -185,7 +186,16 @@ int connect(
       pfd.fd = socket;
       pfd.events = POLLOUT;
 
-      int numReady = ::poll(&pfd, 1, timeout.count());
+      int pollTimeout = timeout.count();
+      if (timeout != kNoTimeout) {
+        // calculate remaining time and use that as timeout for poll()
+        const auto elapsed = std::chrono::high_resolution_clock::now() - start;
+        const auto remaining =
+            std::chrono::duration_cast<std::chrono::milliseconds>(timeout) -
+            std::chrono::duration_cast<std::chrono::milliseconds>(elapsed);
+        pollTimeout = std::max(0, (int)remaining.count());
```

Comment:
have to cast this to int, or else the macOS compiler throws errors.

Reply:
You should cast to int64_t (or use 0LL in std::max()), also use `static_cast<int64_t>` instead of C style casts.